realcc_tests/test_procedure.py

```diff
@@ -0,0 +1,200 @@
+    CattTest("raise volume by 50", ["volumeup", "50"], sleep=3, check_data=("volume_level", 1.0)),
+    CattTest(
+        "cast h264 640x360 / aac content from twitch.tv",
+        ["cast", "-y", "format=360", "https://clips.twitch.tv/CloudyEnticingChickpeaCeilingCat"],
+        check_data=("content_id", "https://clips-media-assets2.twitch.tv/AT-cm%7C304482431-360.mp4"),
+    ),
+    CattTest(
+        "cast h264 1280x720 / aac content directly from google commondatastorage",
+        ["cast", "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"],
+        check_data=("content_id", "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"),
+    ),
+]
+
+YOUTUBE_CTRL_TESTS = [
+    CattTest(
+        "cast playlist from youtube",
+        ["cast", "https://www.youtube.com/watch?list=PLQNHYNv9IpSzzaQMuH7ji2bEy6o8T8Wwn"],
+        check_data=("content_id", "CIvzV5ZdYis"),
+    ),
+    CattTest("skip to next entry in playlist", ["skip"], sleep=15, check_data=("content_id", "Ff_FvEkuG8w")),
+    CattTest(
+        "try to add invalid video-url to playlist (should fail)",
+        ["add", "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"],
+        sleep=3,
+        should_fail=True,
+        check_err="This url cannot be added to the queue.",
+    ),
+]
+
+# Some of the audio tests will fail on non-audio devices,
+# as they are fed the regular video+audio streams.
+AUDIO_ONLY_TESTS = [
+    CattTest(
+        "cast audio-only DASH aac content from facebook",
+        ["cast", "https://www.facebook.com/PixarCars/videos/10158549620120183/"],
+        substring=True,
+        check_data=("content_id", "18106055_10158549666610183_8333687643300691968_n.mp4"),
+    )
+]
+
+STANDARD_TESTS = DEFAULT_CTRL_TESTS + YOUTUBE_CTRL_TESTS
+AUDIO_TESTS = AUDIO_ONLY_TESTS
+ULTRA_TESTS = []  # type: list
+
+
+def run_tests(standard: str = "", audio: str = "", ultra: str = ""):
+    test_outcomes = []  # type: list
+    suites = {}  # type: dict
+    if standard:
+        suites.update({standard: STANDARD_TESTS})
+    if audio:
+        suites.update({audio: AUDIO_TESTS})
+    if ultra:
+        suites.update({ultra: ULTRA_TESTS})
+    if not suites:
+        raise CattTestError("There were no tests to run.")
+
+    for device_name, suite in suites.items():
+        click.secho('Running some tests on "{}".'.format(device_name), fg="magenta")
+        click.secho("------------------------------------------", fg="magenta")
+        cbase = CMD_BASE + [device_name]
+
+        for test in suite:
+            test.set_cmd_base(cbase)
+            click.echo(test.desc + "  ->  ", nl=False)
+            if test.run():
+                click.secho("success!", fg="green")
+                test_outcomes.append(True)
+            else:
+                click.secho("failure!", fg="red")
+                test_outcomes.append(False)
+            if test.dump:
```

Comment:
What is the most sane way of communicating that a test that is supposed to fail, actually fails? The success/failure options seems inadequate here.

Reply:
"Outcome as expected/outcome contrary to expected"? Also "test success/test failure" communicates that the test succeeded, even if the test was to check that something fails, in my opinion.

Reply:
cheers